### PR TITLE
Adds state visibility CIP

### PR DESCRIPTION
**Pull request by systay**
Replaces #6

### CODE REVIEW

File: cip/CIP2015-10-03-State-visibility-between-clauses.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+To allow the user to understand what the outcome will be at an abstract, logical level, semantics for the visibility of these changes should be clear.
+
+=== Proposal
+Each clause lives in its own state, which includes all the changes of clauses coming before it, including changes performed by itself, *but none of the changes made by clauses listed later in the query.*
+
+The semantical data flow here is such that each clause operates on the entire result set before passing control to any subsequent clause.
+
+Updates between UNION subqueries are also treated in a sequential manner.
+Therefore subqueries coming later will see updates from subqueries before, but not the other way around.
+
```
Comment (petraselmer) [inline cip/CIP2015-10-03-State-visibility-between-clauses.adoc]: I think all keywords ought to be formatted, so please change "UNION" -> "`UNION`"

File: cip/CIP2015-10-03-State-visibility-between-clauses.adoc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+==== Examples
+
+.Example query 1:
+The following query would lead to a never-ending loop, if a lazy +MATCH+ clause is not isolated from the new nodes created by +CREATE+.
```
Comment (petraselmer) [inline cip/CIP2015-10-03-State-visibility-between-clauses.adoc]: It seems that "+bla+" doesn't work, as there is no formatting. Rather use backticks.

**Comment by systay** [comment]
Comments shared during CLG meeting:

* Point out explicitly that this is not about concurrency
* Add example with UNION
* Rename this CIP instead of reusing the CIP id

File: cip/CIP2015-10-27-State-visibility-between-clauses.adoc [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+toc::[]
+
+=== Background
+Cypher allows queries to go through multiple steps, and it’s possible to interleave reading and writing to the graph.
+Some clauses can also do both - read the graph and write to it at the same time.
+
+To allow the user to understand what the outcome will be at an abstract, logical level, semantics for the visibility of these changes should be clear.
+
+Note:
+This is not about concurrency - even in a single user system this would need to be thought through and defined explicitly._
```
Comment (thobe) [inline cip/CIP2015-10-27-State-visibility-between-clauses.adoc]: remove trailing underscore

**Pull request merged**